vumi/tests/helpers.py

```diff
@@ -169,6 +171,43 @@ def get_timeout():
     return float(timeout_str)
 
 
+def get_stack_trace(exclude_last=0):
+    """
+    Get a stack trace that can be stored and referred to later.
+
+    The inside of this function is excluded from the stack trace, because it's
+    not relevant. Additionally, all entries prior to the first occurrence of
+    "twisted/trial/_asynctest.py" or "django/test/testcases.py" are removed to
+    avoid unnecessary test runner noise.
+
+    :param int exclude_last:
+        Number of entries to remove from the end of the stack trace. Use this
+        to get rid of wrapper functions or implementation details irrelevant to
+        the purpose of the stack trace.
+
+    :return:
+        A list of strings, each representing a stack frame, in the same format
+        as ``traceback.format_stack()``.
+    """
+    stack = traceback.format_stack()
+
+    def is_boring(entry):
+        return all(pathfrag not in entry for pathfrag in [
+            "twisted/trial/_asynctest.py",
+            "django/test/testcases.py",
+        ])
+
+    filtered_stack = list(dropwhile(is_boring, stack))
+    if filtered_stack:
+        # We haven't accidentally devoured everything.
+        lines_removed = len(stack) - len(filtered_stack)
+        stack = filtered_stack
+        stack.insert(0, "%s test runner lines removed.\n" % lines_removed)
+    # Remove the current stack frame and any extra stack frames we've been
+    # asked to remove.
+    return stack[:-(exclude_last + 1)]
```

Comment:
Wouldn't this mean that the `%s test runner lines removed` line might be removed?